partition_snapshot_row_cursor.hh

```diff
@@ -401,21 +538,43 @@ public:
     // The cursor remains valid after the call and points at the same row as before.
     ensure_result ensure_entry_in_latest() {
         auto&& rows = _snp.version()->partition().mutable_clustered_rows();
-        auto latest_i = get_iterator_in_latest_version();
-        rows_entry& latest = *latest_i;
         if (is_in_latest_version()) {
+            auto latest_i = get_iterator_in_latest_version();
+            rows_entry& latest = *latest_i;
             if (_snp.at_latest_version()) {
                 _snp.tracker()->touch(latest);
             }
             return {latest, false};
         } else {
             // Copy row from older version because rows in evictable versions must
             // hold values which are independently complete to be consistent on eviction.
-            auto e = current_allocator().construct<rows_entry>(_schema, *_current_row[0].it);
-            e->set_continuous(latest_i && latest_i->continuous());
-            _snp.tracker()->insert(*e);
-            rows.insert_before(latest_i, *e);
-            return {*e, true};
+            auto e = [&] {
+                if (!at_a_row()) {
```

Comment:
Why was it not possible before that `_current_row.empty()` and now it is?

Reply:
advance_to() can now advance to a position which is not a row. We need that because in reverse view there is no dummy before all rows (in table order that is, "after all rows" in read order).

See https://github.com/scylladb/scylla/pull/9454#discussion_r763216731